hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -603,6 +615,20 @@ public boolean doMetadataTableValidation() {
         result.add(Pair.of(false, e));
       }
 
+      try {
+        if (cfg.validateSecondaryIndex) {
+          validateSecondaryIndex(engineContext, metadataTableBasedContext, metaClient);
+          result.add(Pair.of(true, null));
+        }
+      } catch (HoodieValidationException e) {
```

Comment:
Is it possible to extract this try-catch block for reusability, e.g. even for partition stats validation it is more or less the same.

Reply:
Addressed. Did not extract out the entire block since the try block varies a lot between the validations.